internal/store/events.go

```diff
@@ -58,54 +67,126 @@ func (s *Service) GetEvents() ([]Event, error) {
 	return events, rows.Err()
 }
 
+// GetEvent retrieves a specific event.
+func (s *Service) GetEvent(eventKey string) (Event, error) {
+	event := Event{Key: eventKey, Location: Location{}}
+	if err := s.db.QueryRow("SELECT name, district, week, start_date, end_date, location_name, lat, lon FROM events WHERE key = $1", eventKey).
+		Scan(&event.Name, &event.District, &event.Week, &event.StartDate, &event.EndDate, &event.Location.Name, &event.Location.Lat, &event.Location.Lon); err != nil {
+		if err == sql.ErrNoRows {
+			return event, NoResultError{err}
+		}
+		if pqError, ok := isPQError(err); ok {
+			return event, getPostgresError(pqError)
+		}
+		return event, err
+	}
+
+	rows, err := s.db.Query("SELECT type, url FROM webcasts WHERE event_key = $1", eventKey)
+	if err != nil {
+		if pqError, ok := isPQError(err); ok {
+			return event, getPostgresError(pqError)
+		}
+		return event, err
+	}
+	defer rows.Close()
+
+	for rows.Next() {
+		var webcast Webcast
+		var webcastType string
+		if err := rows.Scan(&webcastType, &webcast.URL); err != nil {
+			if pqError, ok := isPQError(err); ok {
+				return event, getPostgresError(pqError)
+			}
+			return event, err
+		}
+		webcast.Type = WebcastType(webcastType)
+		event.Webcasts = append(event.Webcasts, webcast)
+	}
+
+	return event, rows.Err()
+}
+
 // EventsUpsert upserts multiple events into the database.
-func (s *Service) EventsUpsert(events []Event) error {
-	eventStmt, err := s.db.Prepare(`
-		INSERT INTO events (id, name, district, week, startDate, endDate, locationName, lat, lon)
+func (s *Service) EventsUpsert(events []Event) (err error) {
+	tx, err := s.db.Begin()
+	if err != nil {
+		return err
+	}
+	defer func() {
```

Comment:
This is *way* too magic and relies specifically on `err` being set. It's pretty clever, but I *really* don't think we should use this.